lib/src/test/java/growthbook/sdk/java/GBFeaturesRepositoryTest.java

```diff
@@ -158,19 +206,54 @@ private static OkHttpClient mockHttpClient(final String serializedBody) throws I
 
         Call remoteCall = mock(Call.class);
 
-        Response response = new Response.Builder()
-            .request(new Request.Builder().url("http://url.com").build())
-            .protocol(Protocol.HTTP_1_1)
-            .code(200).message("").body(
-                ResponseBody.create(
-                    serializedBody,
-                    MediaType.parse("application/json")
-                ))
-            .build();
+        when(remoteCall.execute()).thenReturn(getResponse(serializedBody));
+        when(okHttpClient.newCall(any())).thenReturn(remoteCall);
+
+        //required for original callback okHttpClient
+        doAnswer(invocation -> {
+            ((Callback) invocation.getArgument(0)).onResponse(remoteCall, getResponse(serializedBody));
+            return null;
+        }).when(remoteCall).enqueue(any(Callback.class));
 
-        when(remoteCall.execute()).thenReturn(response);
+        return okHttpClient;
+    }
+
+    /**
+     * Create a mock instance for failed callback of {@link OkHttpClient}
+     *
+     * @param serializedBody JSON string response
+     * @return mock {@link OkHttpClient}
+     */
+    private static OkHttpClient mockHttpClientFailedCallback(final String serializedBody, final IOException ioException) throws IOException {
+        OkHttpClient okHttpClient = mock(OkHttpClient.class);
+        Call remoteCall = mock(Call.class);
+        when(remoteCall.execute()).thenReturn(getResponse(serializedBody));
```

Comment:
It might make sense to pass the status code into `getResponse`—it would make more sense to say that a failure would return an error code instead of 200.

Alternatively, don't abstract `getResponse` (that'll probably make it easier to read) and just do a new failure response inline where an error status is used (and not 200).

Reply:
@tinahollygb hi) so, I rolled back my changes in the test that affects your implementation. Аdded for test testOnFeaturesRefresh_Error failed response.